From 7a4972f36a63eba4d862115265a7bf77ac1e4129 Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.com>
Date: Tue, 7 Jan 2025 12:11:10 +0000
Subject: [PATCH] dts: rp1: PL011 UARTs are actually r1p5

The ARM PL011 UART instances in RP1 are r1p5 spec, which means they
have 32-entry FIFOs. The correct periphid value for this is 0x00341011.
Thanks to N Buchwitz for pointing this out.

Signed-off-by: Phil Elwell <phil@raspberrypi.com>
---
 arch/arm64/boot/dts/broadcom/rp1.dtsi | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

--- a/arch/arm64/boot/dts/broadcom/rp1.dtsi
+++ b/arch/arm64/boot/dts/broadcom/rp1.dtsi
@@ -69,7 +69,7 @@
 			       <&rp1_dma RP1_DMA_UART0_RX>;
 			dma-names = "tx", "rx";
 			pinctrl-names = "default";
-			arm,primecell-periphid = <0x00541011>;
+			arm,primecell-periphid = <0x00341011>;
 			uart-has-rtscts;
 			cts-event-workaround;
 			skip-init;
@@ -86,7 +86,7 @@
 			//        <&rp1_dma RP1_DMA_UART1_RX>;
 			// dma-names = "tx", "rx";
 			pinctrl-names = "default";
-			arm,primecell-periphid = <0x00541011>;
+			arm,primecell-periphid = <0x00341011>;
 			uart-has-rtscts;
 			cts-event-workaround;
 			skip-init;
@@ -103,7 +103,7 @@
 			//        <&rp1_dma RP1_DMA_UART2_RX>;
 			// dma-names = "tx", "rx";
 			pinctrl-names = "default";
-			arm,primecell-periphid = <0x00541011>;
+			arm,primecell-periphid = <0x00341011>;
 			uart-has-rtscts;
 			cts-event-workaround;
 			skip-init;
@@ -120,7 +120,7 @@
 			//        <&rp1_dma RP1_DMA_UART3_RX>;
 			// dma-names = "tx", "rx";
 			pinctrl-names = "default";
-			arm,primecell-periphid = <0x00541011>;
+			arm,primecell-periphid = <0x00341011>;
 			uart-has-rtscts;
 			cts-event-workaround;
 			skip-init;
@@ -137,7 +137,7 @@
 			//        <&rp1_dma RP1_DMA_UART4_RX>;
 			// dma-names = "tx", "rx";
 			pinctrl-names = "default";
-			arm,primecell-periphid = <0x00541011>;
+			arm,primecell-periphid = <0x00341011>;
 			uart-has-rtscts;
 			cts-event-workaround;
 			skip-init;
@@ -154,7 +154,7 @@
 			//        <&rp1_dma RP1_DMA_UART5_RX>;
 			// dma-names = "tx", "rx";
 			pinctrl-names = "default";
-			arm,primecell-periphid = <0x00541011>;
+			arm,primecell-periphid = <0x00341011>;
 			uart-has-rtscts;
 			cts-event-workaround;
 			skip-init;
